packages/manager/apps/dedicated/client/app/ip/components/list/firewall/game/ip-ip-firewall-game.controller.js

```diff
@@ -294,6 +294,66 @@ export default /* @ngInject */ function IpGameFirewallCtrl(
     });
   }
 
+  function hasNewRuleIntoRule(newRule, rule) {
+    return (
+      newRule.ports.from < rule.ports.from && newRule.ports.to < rule.ports.to
+    );
+  }
+
+  function hasNewRulePortToIntoRule(newRule, rule) {
+    return (
+      newRule.ports.from < rule.ports.to &&
+      newRule.ports.to > rule.ports.from &&
+      newRule.ports.to < rule.ports.to
+    );
+  }
+
+  function hasNewRulePortFromIntoRule(newRule, rule) {
+    return (
+      newRule.ports.from > rule.ports.from &&
+      newRule.ports.from < rule.ports.to &&
+      newRule.ports.to > rule.ports.to
+    );
+  }
+
+  function hasRuleIntoNewRule(newRule, rule) {
+    return (
+      newRule.ports.from < rule.ports.from && newRule.ports.to > rule.ports.to
+    );
+  }
+
+  function hasPortsAlreadyUsed() {
+    return self.table.rules.some((rule) => {
+      if (rule.ports.from === rule.ports.to) {
+        if (self.rule.ports.from === self.rule.ports.to) {
+          if (self.rule.ports.from === rule.ports.from) {
+            return true;
+          }
+        } else if (
+          self.rule.ports.from < self.rule.ports.to &&
+          rule.ports.from > self.rule.ports.from
+        ) {
+          return true;
+        }
+      } else if (self.rule.ports.from === self.rule.ports.to) {
+        if (
+          self.rule.ports.from < rule.ports.to &&
+          self.rule.ports.from > rule.ports.from
+        ) {
+          return true;
+        }
+      } else if (
+        hasNewRuleIntoRule(self.rule, rule) ||
+        hasNewRulePortToIntoRule(self.rule, rule) ||
+        hasNewRulePortFromIntoRule(self.rule, rule) ||
+        hasRuleIntoNewRule(self.rule, rule)
+      ) {
+        return true;
+      }
+      return false;
```

Comment:
**nitpick:** Can you add comments to explain what you're doing here. Complicated to understand the business logic.
Is it not possible to simplify the logic here?

Reply:
I added the comments to explain, I should be difficult to simply the logic here.